### PR TITLE
test(payment): INT-4855 increase code coverage on CheckoutcomCustomFo…

### PR DESCRIPTION
**Pull request by JoseAlvarezHdz**
## What?  [INT-4855](https://jira.bigcommerce.com/browse/INT-4855)

Increase CheckoutcomCustomFormFields code coverage

## Why?
To increase code coverage

### Before

Screen Shot 2021-10-29 at 14 56 50
 
### Now

Screen Shot 2021-10-29 at 14 53 58

@bigcommerce/checkout. @bigcommerce/apex-integrations

### CODE REVIEW

**Comment by JoseAlvarezHdz** [comment]
Screen Shot 2021-10-29 at 14 57 55

The only things left to test are 

1. `optional chaining` which i cant test because cant undefined them
2. not defined `_selectedIssuer` but it wont happened as `handleClick` is only trigger when you click on an option so that find will never return an empty object but `find` returns  `T | undefined` so thats why its left

File: src/app/payment/paymentMethod/CheckoutcomCustomFormFields.spec.tsx [hunk 1]
```diff
@@ -78,6 +79,23 @@ const getAPMProps = {
             returnUrl: 'https://test-store.store.bcdev/checkout.php?action=set_external_checkout&provider=checkoutcom',
         },
     }),
+    fawry: () => ({
+        checkoutCustomMethod: 'fawry',
+        method: {
+            if: 'fawry',
```
Comment (FranciscoOnt) [inline src/app/payment/paymentMethod/CheckoutcomCustomFormFields.spec.tsx:85]: shouldn't this be `id`?

File: src/app/payment/paymentMethod/CheckoutcomCustomFormFields.spec.tsx [hunk 2]
```diff
@@ -175,24 +193,57 @@ describe('CheckoutCustomFormFields', () => {
...
+            component = mount(<CheckoutcomAPMsTest { ...idealProps } cardFieldset={ <IdealFormFieldset method={ idealProps.method } /> } />);
+        });
+
+        it('should render the ideal fieldset', () => {
 
-            expect(compoonent.find('input[type="hidden"]')).toHaveLength(1);
+            expect(component.find('input[type="hidden"]')).toHaveLength(1);
+        });
+
+        it('should call handleclick on DropdownTrigger change', () => {
```
Comment (vmparra) [inline src/app/payment/paymentMethod/CheckoutcomCustomFormFields.spec.tsx:240]: Can we please add a description of what  is happening on the UI (i.e. Shopper selects a payment method from dropdown, etc.)

File: src/app/payment/paymentMethod/CheckoutcomCustomFormFields.spec.tsx [hunk 3]
```diff
@@ -206,4 +257,16 @@ describe('CheckoutCustomFormFields', () => {
             expect(compoonent.find('input[name="ccDocument"]')).toHaveLength(1);
         });
     });
+
+    describe('Fawry', () => {
+        const FawryFormFieldset = checkoutcomCustomFormFields.fawry;
+
+        it('should render the Fawry fieldset', () => {
```
Comment (vmparra) [inline src/app/payment/paymentMethod/CheckoutcomCustomFormFields.spec.tsx:264]: Same as above

**Review by vmparra — CHANGES_REQUESTED** [review]
Hi @JoseAlvarezHdz, this is great, thank you for doing it. I left a few comments, please let me know if you have questions.

**Review by FranciscoOnt — APPROVED** [review]
Just a small typo 👍

File: src/app/payment/paymentMethod/CheckoutcomCustomFormFields.spec.tsx (hunk 1 above)
Comment (JoseAlvarezHdz) [reply src/app/payment/paymentMethod/CheckoutcomCustomFormFields.spec.tsx]: Done!

**Review by vmparra — APPROVED** [review]
LGTM 👍🏻

**Review by leeBigCommerce — APPROVED** [review]
LGTM 👍

**Comment by leeBigCommerce** [comment]
@bigcommerce/checkout code owner review please ✨

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**